website/blog/2024-09-19-bootc-microshift.md

```diff
@@ -0,0 +1,284 @@
+
+This entire process is carried out using a single Containerfile (or Dockerfile).
+
+## Requirements
+
+Before starting the tutorial, ensure you have:
+
+- An active [Red Hat account](https://developers.redhat.com/articles/2024/05/07/podman-desktop-red-hat-developer-subscription)
+- Access to an [OpenShift Hybrid Cloud pull secret](https://console.redhat.com/openshift/install/pull-secret)
+- [Podman Desktop installed](https://podman-desktop.io/downloads)
+- [Podman Desktop BootC Extension](https://github.com/containers/podman-desktop-extension-bootc)
+- Your preferred VM-running software (e.g., [using libvirt](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/configuring_and_managing_virtualization/getting-started-with-virtualization-in-rhel-8_configuring-and-managing-virtualization#enabling-virtualization-in-rhel8_virt-getting-started) or [qemu](https://www.qemu.org/))
+
+## Building the BootC container image
+
+First, we'll build the initial BootC container image from which we'll later create a bootable OS.
+
+### Logging into Red Hat registry
+
+Before proceeding, download the Red Hat Authentication extension from the catalog to enable access to Red Hat registries:
+
+![red hat login](img/bootc-microshift/redhat_login.png)
+
+Then log into your account:
+
+![red hat login sign in](img/bootc-microshift/redhat_login2.png)
+
+### Creating the Containerfile (or Dockerfile)
+
+The Containerfile is crucial for creating the bootable image.
+
+It's important to note that we will be providing two arguments during the build: an SSH public key to access the OS post-boot, and your OpenShift Pull Secret to enable MicroShift to pull necessary container images.
+
+Below is the Containerfile we will use, with explanations in the comments:
+
+```Dockerfile
+# Use a BootC supported base image such as rhel-bootc
+FROM registry.redhat.io/rhel9/rhel-bootc:9.4
+
+# For access to your OS after it boots, lets add your SSH public key
+ARG SSH_PUBLIC_KEY
+RUN set -eu; mkdir -p /usr/ssh && \
+  echo 'AuthorizedKeysFile /usr/ssh/%u.keys .ssh/authorized_keys .ssh/authorized_keys2' >> /etc/ssh/sshd_config.d/30-auth-system.conf && \
+  echo ${SSH_PUBLIC_KEY} > /usr/ssh/root.keys && chmod 0600 /usr/ssh/root.keys
+
+# For simplicity, we echo the repo location for to /etc/yum.repos.d/minishift.repo
+RUN echo -e '# OpenShift 4.17 release\n\
+[ocp]\n\
+name=OpenShift 4.17\n\
+baseurl=https://mirror.openshift.com/pub/openshift-v4/$basearch/microshift/ocp/latest-4.17/el9/os/\n\
+enabled=1\n\
+gpgcheck=0\n\
+\n\
+# Dependencies\n\
+[ocp-deps]\n\
+name=OpenShift 4.17 dependencies\n\
+baseurl=https://mirror.openshift.com/pub/openshift-v4/$basearch/dependencies/rpms/4.17-el9-beta/\n\
+enabled=1\n\
+gpgcheck=0' > /etc/yum.repos.d/minishift.repo
```

Comment:
Is there a reason we're pointing to pre-release MicroShift repositories? 
I think we should install a released version of MicroShift as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/image_mode.md?plain=1#L39-L45).